libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -261,10 +261,13 @@ static int AP_Logger_Write(lua_State *L) {
         }
         if (charlen != 0) {
             const char *tmp = luaL_checkstring(L, i);
-            if (strlen(tmp) > charlen) {
+            const size_t slen = strlen(tmp);
+            if (slen > charlen) {
                 return luaL_error(L, "arg %i too long for %c format",i,fmt_cat[i-3]);
             }
-            luaL_addlstring(&buffer, (char *)&tmp, charlen);
+            char tstr[charlen] {};
```

Comment:
Not allowed variable size arrays. It can be 4, 16 or 64. We could make it always 64 I guess.

Reply:
I didn't realize that it was even allowed at all, cool!